agent/php_curl.c

```diff
@@ -113,7 +121,7 @@ static void nr_php_curl_set_default_response_header_callback(
   zval* retval = NULL;
   zval* curlopt = NULL;
 
-  if ((NULL == curlres) || (IS_RESOURCE != Z_TYPE_P(curlres))) {
+  if ((NULL == curlres)) {
```

Comment:
It seems the check became less specific.
Would it make sense to add your great new function here?
`if (!nr_php_is_zval_valid_curl_handle(curlres)) {`

Reply:
Great catch!